src/overlays/actors/ovl_En_Wonder_Talk2/z_en_wonder_talk2.c

```diff
@@ -26,19 +31,274 @@ const ActorInit En_Wonder_Talk2_InitVars = {
+                    // Save Information
+                    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ セーブ情報 \t           %x\n" VT_RST, this->switchFlag);
+                    // Specified message type
+                    osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 指定メッセージ種類     %x\n" VT_RST, this->unk_150);
+                    // Real message type
+                    osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ 実質メッセージ種類     %x\n" VT_RST, this->actor.textId);
+                    // Specified range
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 指定範囲               %d\n" VT_RST, this->actor.posRot.rot.z);
+                    //  Processing range
+                    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 処理範囲               %f\n" VT_RST, this->unk_15C);
+                    // What is your range?
+                    osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ レンジは？ \t\t   %d\n" VT_RST, this->actor.unk_1F);
+                    osSyncPrintf("\n\n");
+                    osSyncPrintf("\n\n");
+                    switch (this->unk_154) {
+                        case 1:
+                            // Compulsion
+                            osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆ 強制 ☆☆ \n" VT_RST);
+                            break;
+                        case 4:
+                            // Geld Training Center Forced Check Only
+                            osSyncPrintf(VT_FGCOL(RED) " ☆☆ ゲルドの修練場強制チェックのみ ☆☆ \n" VT_RST);
+                            break;
+                    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk2/func_80B3A10C.s")
+                    osSyncPrintf("\n\n");
+                }
+                this->unk_158 = 0;
+                if (this->unk_156 == 0) {
+                    func_8010B680(globalCtx, this->actor.textId, NULL);
+                    func_8002DF54(globalCtx, NULL, 8);
+                    this->actor.flags |= 0x11;
+                    this->actionFunc = func_80B3A3D4;
+                    return;
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk2/func_80B3A15C.s")
+            } else {
+                this->unk_156 = 0;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk2/func_80B3A3D4.s")
+void EnWonderTalk2_DoNothing(EnWonderTalk2* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk2/func_80B3A4F8.s")
+void EnWonderTalk2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnWonderTalk2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk2/func_80B3A770.s")
+    this->actionFunc(this, globalCtx);
+    this->actor.posRot.pos.y = this->posRot.y;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Talk2/EnWonderTalk2_Update.s")
+    Actor_SetHeight(&this->actor, this->height);
+    if (BREG(0) != 0) {
+        if (this->unk_158) {
+            if (!(this->unk_158 & 1)) {
```

Comment:
unk_158 seems to be a frame count or timer, not a boolean nor flags.
```suggestion
        if (this->unk_158 != 0) {
            if ((this->unk_158 & 1) == 0) {
```